frontend/src/pages/instance/Editor/index.vue

```diff
@@ -126,7 +127,8 @@ export default {
                 {
                     label: 'Assets',
                     to: { name: 'instance-editor-assets', params: { id: this.instance.id } },
-                    tag: 'instance-assets'
+                    tag: 'instance-assets',
+                    hidden: !this.hasAMinimumTeamRoleOf('member')
```

Comment:
Can we use the enum instead of hardcoded string?

e.g. `RoleNames[Roles.member]`

Better still if `hasAMinimumTeamRoleOf` accepted the actual role enum value e.g. `hasAMinimumTeamRoleOf(Roles.member)`

This way, any refactoring will be caught and references can be found.

Reply:
That was my first intention on how to do it but given that the Roles & Permissions are a domain leak from the backend I'd rather have it contained in a single place in the permissions mixin for the frontend. 

I'd even go as far as remove the other usages of /forge/lib/roles.js from the frontend and have them typehinted for the frontend as it couples the fe to the be. (in another task)

In order to get a bit of both worlds, I added annotations to the method which allows a easy type completion.
![image](https://github.com/user-attachments/assets/7407b3a7-e3a1-4fc2-ac05-a930719bfdd0)

Thinking of this a bit more in on this matter, another way around it would be to preserve the permissions domain leak (as it changes more often and doesn't really make sense to keep two files in sync) and have a copy of the Roles in the frontend to use as an enum (as roles are highly unlikely to change that often)

Of course this is single sided opinion and not set in stone, if you think of a better way of doing it or don't consider the domain leak a problem, i'm all for it.

Reply:
Discussed with @cstns - we agreed to move away from hardcoded strings but that @cstns ideals of domain separation is the right approach (and will be addressed in that regard). Thanks for the insight :)